cmd/ocp-build-data-enforcer/main.go

```diff
@@ -0,0 +1,252 @@
+	var counter int
+	for _, config := range configs {
+		counter++
+		config := config
+		errGroup.Go(func() error {
+			processDockerfile(config)
+			return nil
+		})
+	}
+	if err := errGroup.Wait(); err != nil {
+		logrus.WithError(err).Fatal("Processing failed")
+	}
+	logrus.Infof("Processed %d configs", counter)
+}
+
+func dereferenceStreams(config *ocpImageConfig, streamMap streamMap) {
+	if replacement, hasReplacement := streamMap[config.From.Stream]; hasReplacement && replacement.UpstreamImage != nil {
+		config.From.Stream = *replacement.UpstreamImage
+	}
+	for blder := range config.From.Builder {
+		if replacement, hasReplacement := streamMap[config.From.Builder[blder].Stream]; hasReplacement && replacement.UpstreamImage != nil {
+			config.From.Builder[blder].Stream = *replacement.UpstreamImage
+		}
+	}
+}
+
+func processDockerfile(config ocpImageConfig) {
+	orgRepo := config.orgRepo()
+	log := logrus.WithField("file", config.SourceFileName).WithField("org/repo", orgRepo)
+	split := strings.Split(orgRepo, "/")
+	if n := len(split); n != 2 {
+		log.Errorf("splitting orgRepo didn't yield 2 but %d results", n)
+		return
+	}
+	if split[0] == "openshift-priv" {
+		log.Warn("Ignoring repo in openshift-priv org")
+		return
+	}
+	org, repo := split[0], split[1]
+	getter := github.FileGetterFactory(org, repo, "release-4.6")
+
+	log = log.WithField("dockerfile", config.dockerfile())
+	data, err := getter(config.dockerfile())
+	if err != nil {
+		log.WithError(err).Error("Failed to get dockerfile")
+		return
+	}
+	if len(data) == 0 {
+		log.Error("dockerfile is empty")
+		return
+	}
+
+	updated, hasDiff, err := updateDockerfile(data, config)
+	if err != nil {
+		log.WithError(err).Error("Failed to update Dockerfile")
+		return
+	}
+	if !hasDiff {
+		return
+	}
+	diff := difflib.UnifiedDiff{
+		A:        difflib.SplitLines(string(data)),
+		B:        difflib.SplitLines(string(updated)),
+		FromFile: "original",
+		ToFile:   "updated",
+		Context:  3,
+	}
+	diffStr, err := difflib.GetUnifiedDiffString(diff)
+	if err != nil {
+		log.WithError(err).Error("Failed to construct diff")
+	}
+	log.Infof("Diff:\n---\n%s\n---\n", diffStr)
+}
+
+func readStreamMap(ocpBuildDataDir string) (streamMap, error) {
+	path := filepath.Join(ocpBuildDataDir, "streams.yml")
+	data, err := ioutil.ReadFile(path)
+	if err != nil {
+		return nil, fmt.Errorf("failed to read %s: %w", path, err)
+	}
+	streamMap := streamMap{}
+	if err := yaml.Unmarshal(data, &streamMap); err != nil {
+		return nil, fmt.Errorf("failed to unmarshal %s into streamMap: %w", path, err)
+	}
+	streamMap.defaultImages()
+	return streamMap, nil
+}
+
+func gatherAllOCPImageConfigs(ocpBuildDataDir string) ([]ocpImageConfig, error) {
+	var result []ocpImageConfig
+	resultLock := &sync.Mutex{}
+	errGroup := &errgroup.Group{}
+
+	path := filepath.Join(ocpBuildDataDir, "images")
+	if err := filepath.Walk(path, func(path string, info os.FileInfo, err error) error {
+		if err != nil {
```

Comment:
You already checking that outside of the walk, right? meaning in 172 line.

Reply:
From `filepath.Walks` docs:

```
All errors that arise visiting files
and directories are filtered by walkFn
```

Which means `WalkFN` must return it or it will get swallowed.